clientconn.go

```diff
@@ -118,6 +118,15 @@ func WithPicker(p Picker) DialOption {
 // WithBackoffConfig configures the dialer to use the provided backoff
 // parameters after connection failures.
 func WithBackoffConfig(b *BackoffConfig) DialOption {
```

Comment:
`WithBackoffConfig(&BackoffConfig{MaxDelay: time.Second})` is invalid without this fix. It is only a partial config. Your proposal is fine, but I don't see why fixing this requires creating another exported function.